peps/pep-0667.rst

```diff
@@ -31,7 +30,30 @@ consistent regardless of threading or coroutines.
 
 The ``locals()`` function will act the same as it does now for class
 and modules scopes. For function scopes it will return an instantaneous
-snapshot of the underlying ``frame.f_locals``.
+snapshot of the underlying ``frame.f_locals`` rather than implicitly
+refreshing a single shared dictionary cached on the frame object.
+
+Implementation Notes
+====================
+
+When accepted, the PEP text suggested that ``PyEval_GetLocals`` would start returning a
+cached instance of the new write-through proxy, while the implementation sketch indicated
+it would continue to return a dictionary snapshot cached on the frame instance. This
+discrepancy was identified while implementing the PEP, and
+`resolved by the Steering Council <https://github.com/python/steering-council/issues/245#issuecomment-2179005461>`__
+in favour of retaining the Python 3.12 behaviour of returning a dictionary snapshot
+cached on the frame instance.
+
+To avoid confusion when following the reference link from the Python 3.13 What's New
+documentation, the PEP text has been updated accordingly.
+
+During the discussions of the C API clarification, it also became apparent that the
+rationale behind ``locals()`` being updated to return independent snapshots in optimized
```

Comment:
Do you think "optimized scopes" needs some additional definition or context?  It's terminology used often in the PEP but undefined.  Perhaps just add a link to [the glossary entry](https://docs.python.org/3.13/glossary.html#term-optimized-scope)?

Reply:
Good idea, I'll add the cross-reference.